### ISSUE TITLE
Missing file

### ISSUE THREAD

**Issue by lichennan123** [wang-lab/IMAPR#2]
Hello!

I have been using this tool lately to identify mutations from bulk RNAseq data. My process got stuck at a step when a file named "extract_tmp/IDs_all.txt" was not found, which was the first error that it encountered. I checked your source code "detect_variants.pl" and noticed that there was not a prior command that generated this file. The chunk of codes relevant to this is attached below. I have no idea how to solve this issue. Do you have any idea how we could troubleshoot this? Thank you.

`open(my $IDs_all, ">", "$tmp_dir/IDs_all.txt") or die "Cannot open IDs_all.txt: $!";`
...
`system("java -Xmx7g -jar $picard FilterSamReads -I $bamInput -O $tmp_dir/tmp_bam.bam -READ_LIST_FILE $tmp_dir/IDs_all.txt -FILTER includeReadList -WRITE_READS_FILES false -VALIDATION_STRINGENCY LENIENT 2>> $hisat2_log");
`
The error was:
`ERROR	2024-08-29 01:57:56	FilterSamReads	Failed to filter Lym1.STAR.bam
htsjdk.samtools.SAMException: /data/lic27/project_clon_evo/lym1-rnaseq/05-rna-mutcall/extract_tmp/IDs_all.txt has length 0`

Sincerely,
Chennan

**Comment by leagleag** [comment]
Just to confirm I am hitting the same error.

**Comment by lichennan123** [comment]
@leagleag I do not believe I figured out the solution. Instead, I followed the GATK best practice - https://gatk.broadinstitute.org/hc/en-us/articles/360035531192-RNAseq-short-variant-discovery-SNPs-Indels

**Comment by leagleag** [comment]
Thanks for your reply. Empty bed and mutect files were the problem.
